rio/subsystems/drivetrain.py

```diff
@@ -174,21 +243,197 @@ def setModuleState(self, newState):
+
+        # If the current swerve module is the fp module then move the fp encoder
+        if id == 6:
+
+            # If the encoder is close enough to the target, dont move the encoder
+            # NOTE: Doesent work very well
+            if (
+                self.sim_fp_encoder > target - 0.1
+                and self.sim_fp_encoder < target + 0.1
+            ):
+                pass
+
+            # If the encoder is below the target, move the encoder value up
+            elif self.sim_fp_encoder < target:
+                self.sim_fp_encoder = self.sim_fp_encoder + (8.3 * 0.05)
+
+            # If the encoder is above the target, move the encoder value down
+            else:
+                self.sim_fp_encoder = self.sim_fp_encoder - (8.3 * 0.05)
+
+            # Return the encoder value
+            return self.sim_fp_encoder
+
+        # If the current swerve module is the fs module then move the fs encoder
+        elif id == 1:
+
+            # If the encoder is close enough to the target, dont move the encoder
+            # NOTE: Doesent work very well
+            if (
+                self.sim_fs_encoder > target - 0.1
+                and self.sim_fs_encoder < target + 0.1
+            ):
+                pass
+
+            # If the encoder is below the target, move the encoder value up
+            elif self.sim_fs_encoder < target:
+                self.sim_fs_encoder = self.sim_fs_encoder + (8.3 * 0.05)
+
+            # If the encoder is above the target, move the encoder value down
+            else:
+                self.sim_fs_encoder = self.sim_fs_encoder - (8.3 * 0.05)
+
+            # Return the encoder value
+            return self.sim_fs_encoder
+
+        # If the current swerve module is the as module then move the as encoder
+        elif id == 4:
+
+            # If the encoder is close enough to the target, dont move the encoder
+            # NOTE: Doesent work very well
+            if (
+                self.sim_as_encoder > target - 0.1
+                and self.sim_as_encoder < target + 0.1
+            ):
+                pass
+
+            # If the encoder is below the target, move the encoder value up
+            elif self.sim_as_encoder < target:
+                self.sim_as_encoder = self.sim_as_encoder + (8.3 * 0.05)
+
+            # If the encoder is above the target, move the encoder value down
+            else:
+                self.sim_as_encoder = self.sim_as_encoder - (8.3 * 0.05)
+
+            # Return the encoder value
+            return self.sim_as_encoder
+
+        # If the current swerve module is the ap module then move the ap encoder
+        elif id == 8:
+
+            # If the encoder is close enough to the target, dont move the encoder
+            # NOTE: Doesent work very well
+            if (
+                self.sim_ap_encoder > target - 0.1
+                and self.sim_ap_encoder < target + 0.1
+            ):
+                pass
```

Comment:
I cant even select all this code to comment on it, this should deffo be in `physics.py`...